### DISCUSSION TITLE
Custom OIDC auth not offered in Login page

### DISCUSSION THREAD

**Discussion by borrinimarco74** [outline/outline#2656] [Self Hosting]
While attempting to use an external IdP (keycloak with google support) to manage the situation where I'm not able to use at the same time Google Signin with:

- my corporate email domain (managed by Google)
- some external domains (managed by Google)
- GMail accounts

in a self hosted installation (for me, the infamous **_Sorry, Google sign in cannot be used with a personal email. Please try signing in with your Google Workspace account._** error for eveything that it's not a corporate email address, see #https://github.com/outline/outline/discussions/2581),

I'm now stuck in Outline authentication page: I'm not offered the choice to use something different than Gmail SignIn 

**To Reproduce**
1. In .env configure GOOGLE_CLIENT_* and OIDC_* fields

**Expected behavior**
A choice beetween Google SignIn and custom OIDC auth

**Screenshots**

**Outline:**
- Install: self hosted (Docker)
- Version: 0.59

**Desktop:**
 - OS: Oracle Linux 8
 - Browser Firefox
 - Version: 78.14.0

[Update]
The issue seems to be related to the fact that once a user is authenticated against an authentication backend, everything else is not offered. Just tested with a clean development environment.

**Comment by tommoor** [comment]
It's unusual to set two login options on self-hosted, this might be a bug that once a team is created it will default to showing sign-in options for that team

**Reply by borrinimarco74** [reply]
It's clear that it's unusual to offer two login options for a (docker-ized) self hosted solution but in my scenario (a single team joined by every account) it's a way to try to bypass a much bigger problem.

I'm trying to migrate a Google Sites wiki where accounts are represented by internal corporate users (email accounts managed by Google), external users (let's say gmail.com + various external email address managed by Google).

My first access created the default team; each following corporate user joins the default team just created and everything works fine BUT every other access always gives me "Sorry, Google sign in cannot be used with a personal email. Please try signing in with your Google Workspace account."

I'm not able to fix this issue so I've tried tried different approaches:

- I've added Keycloak as external IdP in my docker environment; I can auth against it with every google email address but when I fill OIDC_* values in .env I cannot see anything in login page (only google authentication). If I also disable Google auth in .env then I see no login options at all...
- I've tried to enable Email access after adding a (working) postfix image in my docker environment by filling SMTP_* values or to invite people: nothing happens... Outline really makes no TCP connection to the configured SMTP server in .env

At the end I would be very happy to use only the Google auth if I could avoid the infamous "Google sign in cannot be used with a personal email...".

ps
I know, I've been very verbose

**Reply by borrinimarco74** [reply]
Just updated main post, in my opinion it could be a bug.

**Reply by borrinimarco74** [reply]
> It's unusual to set two login options on self-hosted, this might be a bug that once a team is created it will default to showing sign-in options for that team

Just to understand: If it could be possible to offer two choices, the output of `select * from authentication_providers` should return two lines?

**Comment by michaelhofer** [comment]
I have exactly the same issue. I can see that the issue #2653 has been closed, but can't see the discussion around it. Has this been fixed or was this a configuration issue?
We run the latest stable version (0.59.0)

**Reply by borrinimarco74** [reply]
The issue has been closed and put into a discussion here.

**Comment by vc43** [comment]
I have the same problem. We want to migrate from Slack to OIDC.

**Comment by Limezy** [comment]
Did somebody find a way to show two different identification methods already ?
I don't mind building outline if there is a patch needed.
Thanks !

**Reply by Limezy** [reply]
My bad, @tommoor has been working on it already : https://github.com/outline/outline/pull/2986
